packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -424,6 +448,149 @@ void _cleanUpBitmapConversionCaches() {
   _bitmapBlobUrlCache.clear();
 }
 
+/// Converts a [BitmapDescriptor] into a [Node] that can be used as
+/// [AdvancedMarker]'s icon.
+Future<web.Node?> _advancedMarkerIconFromBitmapDescriptor(
+  BitmapDescriptor bitmapDescriptor, {
+  required double? opacity,
+  required bool isVisible,
+  required double? rotation,
+}) async {
+  if (bitmapDescriptor is PinConfig) {
+    final gmaps.PinElementOptions options =
+        gmaps.PinElementOptions()
+          ..background =
+              bitmapDescriptor.backgroundColor != null
+                  ? _getCssColor(bitmapDescriptor.backgroundColor!)
+                  : null
+          ..borderColor =
+              bitmapDescriptor.borderColor != null
+                  ? _getCssColor(bitmapDescriptor.borderColor!)
+                  : null;
+
+    final AdvancedMarkerGlyph? glyph = bitmapDescriptor.glyph;
+    switch (glyph) {
+      case final CircleGlyph circleGlyph:
+        options.glyphColor = _getCssColor(circleGlyph.color);
+      case final TextGlyph textGlyph:
+        final web.Element element = web.document.createElement('p');
+        element.text = textGlyph.text;
+        if (textGlyph.textColor != null) {
+          element.setAttribute(
+            'style',
+            'color: ${_getCssColor(textGlyph.textColor!)}',
+          );
+        }
+        options.glyph = element;
+      case final BitmapGlyph bitmapGlyph:
+        final web.Node?
+        glyphBitmap = await _advancedMarkerIconFromBitmapDescriptor(
+          bitmapGlyph.bitmap,
+          // Always opaque, opacity is handled by the parent marker.
+          opacity: 1.0,
+          // Always visible, as the visibility is handled by the parent marker.
+          isVisible: true,
+          rotation: rotation,
+        );
+        options.glyph = glyphBitmap;
+      case null:
+        break;
+    }
+
+    final gmaps.PinElement pinElement = gmaps.PinElement(options);
+    final web.HTMLElement htmlElement = pinElement.element;
+    htmlElement.style
+      ..visibility = isVisible ? 'visible' : 'hidden'
+      ..opacity = opacity?.toString() ?? '1.0'
+      ..transform = rotation != null ? 'rotate(${rotation}deg)' : '';
+    return htmlElement;
+  }
+
+  if (bitmapDescriptor is MapBitmap) {
+    final String url = switch (bitmapDescriptor) {
+      (final BytesMapBitmap bytesMapBitmap) => _bitmapBlobUrlCache.putIfAbsent(
+        bytesMapBitmap.byteData.hashCode,
+        () {
+          final web.Blob blob = web.Blob(
+            <JSUint8Array>[bytesMapBitmap.byteData.toJS].toJS,
+          );
+          return web.URL.createObjectURL(blob as JSObject);
+        },
+      ),
+      (final AssetMapBitmap assetMapBitmap) => ui_web.assetManager.getAssetUrl(
+        assetMapBitmap.assetName,
+      ),
+      _ => throw UnimplementedError(),
+    };
+
+    final web.Element icon = web.document.createElement('img')
+      ..setAttribute('src', url);
```

Comment:
Let's use a more specific type so we can use `icon.style`:
```suggestion
    final web.HTMLImageElement icon = web.HTMLImageElement()..src = url;
```